src/app/commandlinecontroller.h

```diff
@@ -25,37 +25,64 @@
 #include <QCommandLineParser>
 #include <QStringList>
 
-#include "modularity/ioc.h"
 #include "global/iapplication.h"
-#include "ui/iuiconfiguration.h"
-#include "importexport/imagesexport/iimagesexportconfiguration.h"
-#include "importexport/midi/imidiconfiguration.h"
-#include "importexport/audioexport/iaudioexportconfiguration.h"
-#include "importexport/videoexport/ivideoexportconfiguration.h"
-#include "appshell/iappshellconfiguration.h"
-#include "appshell/internal/istartupscenario.h"
-#include "notation/inotationconfiguration.h"
-#include "project/iprojectconfiguration.h"
-#include "importexport/guitarpro/iguitarproconfiguration.h"
+#include "io/path.h"
 
 namespace mu::app {
 class CommandLineController
 {
-    INJECT(appshell, framework::IApplication, application)
-    INJECT(appshell, ui::IUiConfiguration, uiConfiguration)
-    INJECT(appshell, iex::imagesexport::IImagesExportConfiguration, imagesExportConfiguration)
-    INJECT(appshell, iex::midi::IMidiImportExportConfiguration, midiImportExportConfiguration)
-    INJECT(appshell, iex::audioexport::IAudioExportConfiguration, audioExportConfiguration)
-    INJECT(appshell, iex::videoexport::IVideoExportConfiguration, videoExportConfiguration)
-    INJECT(appshell, appshell::IAppShellConfiguration, configuration)
-    INJECT(appshell, appshell::IStartupScenario, startupScenario)
-    INJECT(appshell, notation::INotationConfiguration, notationConfiguration)
-    INJECT(appshell, project::IProjectConfiguration, projectConfiguration)
-    INJECT(appshell, iex::guitarpro::IGuitarProConfiguration, guitarProConfiguration);
-
+    //! NOTE: This controller is created at the earliest stage of the application initialization
+    //! You should not inject anything into it
 public:
     CommandLineController() = default;
 
+    struct UiConfig {
```

Comment:
It doesn't matter, but I would make one structure like this

```
    struct Options {
        struct {
            std::optional<int> physicalDotsPerInch;
        } ui;
        
        struct {
            std::optional<bool> fullMigration;
        } project;
    };
    
    Options m_options;
    
    
    m_options.ui.physicalDotsPerInch = 42;
    m_options.project.fullMigration = true;
```

Reply:
Yes, looks much better, thanks!